nbbuild/antsrc/org/netbeans/nbbuild/GitBranchHash.java

```diff
@@ -79,31 +79,40 @@ public void execute() throws BuildException {
             if (headroot != null && Files.size(headroot) > 0l) {
                 List<String> lines = Files.readAllLines(headroot);
                 String line = lines.get(0);
-                String revLink = line.substring(line.indexOf(':')+1).trim();
-                branch = revLink.substring(revLink.lastIndexOf('/')+1).trim();
-                Path revPath = headroot.getParent().resolve(revLink);
-                if(Files.isRegularFile(revPath) && Files.size(revPath) > 0l) {
-                    List<String> revlines = Files.readAllLines(revPath);
-                    String revline = revlines.get(0);
-                    if(revline.length()>=12){
-                        hash = revline.trim();
+                if (!line.contains(":") && (line.length() == 40)) {
```

Comment:
Not sure if this should be `>= 40` and also trimmed? I have also no idea if `>= 12` was just copying from HgID or necessary. :shrug:

Reply:
I do not know that either. Just made some adjustments to work with a detached head.